asset-transfer-basic/rest-api-typescript/src/server.ts

```diff
@@ -13,13 +13,19 @@ import { healthRouter } from './health.router';
 import { jobsRouter } from './jobs.router';
 import { logger } from './logger';
 import { transactionsRouter } from './transactions.router';
+import cors from 'cors';
+import { devCorsEnabled } from './config';
 
 const { BAD_REQUEST, INTERNAL_SERVER_ERROR, NOT_FOUND } = StatusCodes;
 
 export const createServer = async (): Promise<Application> => {
   const app = express();
 
   // Remember for production usage, to check any TLS or CORS requirements
+  // this wildcard option *MIGHT* not be applicable to your environment
+  if (devCorsEnabled) {
+    app.use(cors());
```

Comment:
The wildcard option still doesn't seem right- is `localhost` a possibility? Also, I think some extra config is required since the sample makes use of an `X-API-Key` header- is that enough to require preflight to be enabled?
>Certain CORS requests are considered 'complex' and require an initial OPTIONS request (called the "pre-flight request"). An example of a 'complex' CORS request is one that uses an HTTP verb other than GET/HEAD/POST (such as DELETE) or that uses custom headers.

Reply:
can't say for certain what the theory states, but I was testing with all the HTTP verbs on offer and it was working :-)

Reply:
Well if it works on your machine... :)